### ISSUE TITLE
dockerTools.pullImage does not handle "/"

### ISSUE THREAD

**Issue by fubarbaz** [NixOS/nixpkgs#20435]
I am using the expression `testImage` as the right hand side of a `fromImage` of dockerTools.pullImage.

```
 testImage = dockerTools.pullImage {
    imageName = "nightscape/nix-in-docker"; 
    sha256 = "78029a6894d6778e880a2e37e99b0e5503f5e99cf1cfc11935160add2faa3487";
};
```
I am using master, but this also is the case for 16.09. I am running NixOS 16.09. 

I did `docker pull nightscape/nix-in-docker` (exit status 0). 

Then I expected that `nix-build file.nix` would use the docker image as downloaded, but instead I got: 

```
error: invalid character ‘/’ in name ‘nightscape/nix-in-docker-latest’
```
This error is generated within Nix: (the package manager) 
```
src/libstore/store-api.cc
95:            throw Error(format("invalid character ‘%1%’ in name ‘%2%’")
```

Expected behaviour is that I can fill in any string no the right handside of `imageName`.

**Comment by domenkozar** [comment]
@lethalman should we split the imageName?

**Comment by lethalman** [comment]
Uhm didn't that docker images could have a slash. Yes we should replace / with something like a dash for the nix store path. I believe a bunch of other things would break as well.

**Comment by PierreR** [comment]
I wish for a fix for this issue as it has just hit me.

The example provided in the nixpkgs source does not work: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/docker/examples.nix#L85

**Comment by PierreR** [comment]
This is actually fixed by 8d3cb0af9e24b867df45fc7f35eddbf3e4449404 but it did not make it for `17.03` apparently.

@domenkozar closing the issue might be less deceptive ;-)

**Comment by roberth** [comment]
This has been fixed. If it reoccurs please open a new issue referencing this one.
